syntax = "proto3";
package packets;

enum PacketStateCode {
    ACTIONSTATE = 0;
}

enum ActionPacketCode {
    ACTION = 0;
}

enum PacketCode {
    ENTERWORLD = 0;
    ENTERPACKET = 1;
    MOVE = 2;
    LEAVEWORLD = 3;
    LEAVEPACKET = 4;
	LOGINREQ = 5;
	LOGINACK = 6;
	CHARSELECTREQ = 7;
	CHARSELECTACK = 8;
	PINGREQ = 9;
	UPDATEDISPATCHREQ = 10;
	VSDISCONNECTREQ = 11;
	VSDISCONNECTNOTIFY = 12;
}

enum ActorType {
    PC = 0;
    NPC = 1;
}

enum ActionType {
    STAND = 0;
    WALK = 1;
    RUN = 2;

    // ETC
}

enum ResultType {
    RESULT_SUCCESS = 0;
    RESULT_FAILED = 1;
}

message Vector3 {
    float X = 1;
    float Y = 2;
    float Z = 3;
}